javaharness/java/arcs/android/impl/AndroidHarnessController.java

```diff
@@ -8,22 +8,30 @@
 import arcs.api.ArcsEnvironment;
 import arcs.api.DeviceClient;
 import arcs.api.HarnessController;
+import arcs.api.RuntimeSettings;
+import java.util.logging.Logger;
 import javax.inject.Inject;
+import javax.inject.Provider;
 
 public class AndroidHarnessController implements HarnessController {
+  private static final Logger logger =
+      Logger.getLogger(AndroidHarnessController.class.getName());
 
   private ArcsEnvironment environment;
   private DeviceClient deviceClient;
   private WebView webView;
+  private Provider<RuntimeSettings> runtimeSettings;
 
   @Inject
   AndroidHarnessController(
       ArcsEnvironment environment,
       DeviceClient deviceClient,
-      WebView webView) {
+      WebView webView,
+      Provider<RuntimeSettings> runtimeSettings) {
```

Comment:
Why use a Provider here?

Reply:
Good question, Provider gives us the ability of "re-resolve the dependency".
In this case, it gives us the ability of "re-fetch the property values" at run-time.

So, each time you use obj.get() that guarantees the android properties are re-fetched and all changes are reflected properly.

Reply:
Explained.